packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -33,29 +34,24 @@ export interface AwsS3MultipartOptions extends PluginOptions {
     ) => MaybePromise<{ uploadId: string; key: string }>
     listParts?: (
       file: UppyFile,
-      opts: { uploadId: string; key: string; signal: AbortSignal }
```

Comment:
`signal` is used in several of the typescript apis but only documented once on the webpage. I assume that the web documents are correct.

Reply:
Can you clarify what you mean here by webpage? Do you mean the docs on uppy.io? The types should probably match the code, not the documentation (the documentation should also match the code, but that's a separate discussion)

Reply:
For 'webpage' I am refering to the pages

https://uppy.io/docs/aws-s3/#options
https://uppy.io/docs/aws-s3-multipart/#options
https://uppy.io/docs/xhr-upload/#options

as relevant. Note that I did not check xhr-upload's types beyond using that link to define `AwsS3`'s `options` parameter; which links to the `XHR` documentation.

Reply:
I will see about looking into the code to see what sort of options are actually being used. Looking at a third variable beyond the documentation & the types seemed to be a significantly harder issue however. Thus I did the pr under the assumption the code was written to a design spelled out in the documentation versus the admittedly much more likely (and ultimately correct) paradigm of the documentation being written off the preexisting code.

Reply:
That's totally fair, and btw it's absolutely OK if you don't want to pursue that and prefer I take it to the finish line. I wish that code and documentation would align, but it always drifts apart, so I wouldn't rely solely on it to update the types.

Reply:
I took a look at this, starting with `aws-s3`'s `getUploadParameters`. The code does allow for `method`, `fields` and `headers` to be optional; requiring only `url`. However I was not able to better identify the types. 

For example `method` appears to allow for more then just `'POST' | 'PUT'` starting with allowing any capitalization of those strings (ie: `pOsT` seems like it might be legit). Even if that was the case however as far as declared type information goes i think having stricter types then actually needed might not be a bad thing.

`fields` and `headers` i traced out a bit; where they get inserted into the data store. However I did not see any type information on that data store (and it may not exist as the project is fundamentally in base javascript over typescript). Finding all possible *uses* of the data in the data store and how they desire type information was not something I had the time for.

I would expect other apis to give me the same sorts of issues identifying type information from the code. Unfortunately I do not think I have time to fully trace out the type information on the actual code. Especially given a possible assumption that the documentation may not actually be correct.  Thus if you want to take over that would be appreciated.

Reply:
We do pass the signal here:
https://github.com/transloadit/uppy/blob/fa19db7858069a57046c5e731921d07c01e46ded/packages/@uppy/aws-s3-multipart/src/index.js#L319